src/megaclient.cpp

```diff
@@ -8970,9 +9190,44 @@ bool MegaClient::startxfer(direction_t d, File* f, bool skipdupes)
         }
         else
         {
-            t = new Transfer(this, d);
-            *(FileFingerprint*)t = *(FileFingerprint*)f;
-            t->size = f->size;
+            it = cachedtransfers[d].find(f);
+            if (it != cachedtransfers[d].end())
+            {
+                LOG_debug << "Resumable transfer detected";
+                FileAccess* fa = fsaccess->newfileaccess();
+                if (fa->fopen(&it->second->localfilename)
```

Comment:
Check whether the fingerprint has not changed upon resumption of upload transfers.